### DISCUSSION TITLE
.odesign file into sirius-web

### DISCUSSION THREAD

**Discussion by belaboda-wecan** [eclipse-sirius/sirius-components#3946] [Q&A] [answered]
Hi,

Please, guide me to the right direction on bringing my existing desktop .odesign file and meta-model into Sirius-web.

I guess, the documentation lacks such an example, and as far I see. There are only odesigns defined by API in Sirius-web starter projects.
Is it still available to import .odesign files with ISiriusConfiguration? Sadly, I can not make it work in the latest version.

Thanks for your guidance in advance.

**Comment by belaboda-wecan** [comment]
Managed to do it with a template initializer reading with SiriusDesktopRepresentationDescriptionProvider from compatibility:
```
@Configuration
@ComponentScan(basePackages = {
	    "org.eclipse.sirius.components.compatibility",
	    "org.eclipse.sirius.components.compatibility.emf"
	})

public class FooProjectTemplatesInitializer implements IProjectTemplateInitializer {
    
    private final SiriusDesktopRepresentationDescriptionProvider descriptionProvider;
	
    private final IDiagramCreationService diagramCreationService;

    private final IRepresentationPersistenceService representationPersistenceService;
...
}

```
And an ISiriusConfiguration bean like this:
```
@Configuration
public class FooStarterConfiguration implements ISiriusConfiguration{
			
	@Override
	public List<String> getODesignPaths() {
		return List.of("description/foo.odesign");
	}			
```

But still, I need to implement an IIdentifierProvider, otherwise I get a bean not found error:
```
Parameter 0 of constructor in org.eclipse.sirius.components.compatibility.services.diagrams.ToolProvider required a bean
of type 'org.eclipse.sirius.components.compatibility.api.IIdentifierProvider' that could not be found.
```
Can anyone help with an IIdentifierProvider example implementation or what its role is?
(Strangely this interface is implemented in diverse-project, but it seems way out dated and my gut feeling is that it should be done depending on this diverse-project)

Thanks for the help again.

**Comment by pcdavid** [comment, marked as answer]
Hi.

The Sirius Desktop `odesign` files are no longer supported. Parts of the compatibility code is still there in the repo, but it is not used in the new (DDD-based) architecture and will probably be removed. 

The path forward is to use the View DSL (metamodel) to define your modelers. It's quite similar in the general principle (EMF based, uses concepts like _Node Description_ which corresponds to _Node Mappings_ in Sirius Desktop, AQL-based expressions extensible with custom Java services, etc.).

You can define your modeler "manually" in the web UI if it is small/simple enough, otherwise the recommended way is to use the Java APIs we provide (on top of the plain EMF APIs for the metamodel) in the `sirius-components-view-builder` module to produce the View model programatically.

 [`sirius-web-papaya`](https://github.com/eclipse-sirius/sirius-web/tree/master/packages/sirius-web/backend/sirius-web-papaya) is a good example of a non trivial studio defined this way, see for example [`ClassDiagramDescriptionProvider`](https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/classdiagram/ClassDiagramDescriptionProvider.java).

[Eclipse SysON](https://github.com/eclipse-syson/syson) also follows this approach but is more complex and includes its own abstractions on top.

**Reply by belaboda-wecan** [reply]
Thanks for the detailed explanation. Sorry to hear that. You confirmed my impression too when I first looked at the examples.
